python_utils_test.py

```diff
@@ -0,0 +1,133 @@
+# coding: utf-8
+#
+# Copyright 2019 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for feature detection utilities for Python 2 and Python 3."""
+
+import StringIO
+import ast
+import io
+import sys
+import tempfile
+
+# pylint: disable=relative-import
+from core.tests import test_utils
+from core.tests.data import unicode_and_str_handler
+import python_utils
+# pylint: enable=relative-import
+
+
+class PythonUtilsTests(test_utils.GenericTestBase):
```

Comment:
This test is pretty complicated. Please split the class into two, one for 2 and the other for 3. We can conditionally disable them base on the python-version, and can have a base-class for test methods that should be shared by both.

Reply:
Done